lib/Backend/amd64/LowererMDArch.h

```diff
@@ -29,7 +29,61 @@ class LowererMDArch
     int                 helperCallArgsCount;
     IR::Opnd *          helperCallArgs[MaxArgumentsToHelper];
 
-
+#ifndef _WIN32
+    class XPlatRegArgList
+    {
+        bool isRecording;
+        int lastPosition;
+    public:
+        XPlatRegArgList():isRecording(false) { Reset(); }
+        inline void Reset() { floatCount = 0; intCount = 0; lastPosition = 0;}
+
+        inline void StartRecording() { Assert(!isRecording); Reset(); isRecording = true; }
+        
+        void StopRecording()
+        {
+            Assert(isRecording);
+            isRecording = false;
+            for (int i = lastPosition - GetArgsCount(); i >= 0; i--) args[i] = TyMachPtr;
+        }
+        
+        inline bool IsRecording() { return isRecording; }
+
+        inline int  GetArgsCount() { return intCount + floatCount; }
+        
+        inline int  GetTopPosition() { return lastPosition; }
```

Comment:
Personal preference, as I think it is redundant and gets in the way of code reading.